client/api/src/light.rs

```diff
@@ -82,11 +82,6 @@ pub struct RemoteReadChildRequest<Header: HeaderT> {
 	pub header: Header,
 	/// Storage key for child.
 	pub storage_key: Vec<u8>,
```

Comment:
I think the first option is good (I obviously prefer second option as I wrote it first but the point of this PR is to avoid the general api and switch to the specific one), will need an update on polkadot js too.

Reply:
hmm but now you have storage_key and child_type, shouldn't it be child_info and child_type ? I mean if child_type is not `ParentKeyId` then there might be information missing to be able to answer the request, no ?

Reply:
It comes to the idea that there is no way information would not be accessible in state.
Here child type is only a way to avoid information redundancy in the storage key and not a way to have additional information in the api (as would be child_info).
So with this api child_info, when needed, should be resolve from the system (even if it can not be easy to implement). So this makes the previous default child type impossible to implement, and force a managed unique id approach in this case (from a specification perspective it was not really simple to legitimate child_info, only argument was that managed unique id did not fit the timeline but considering the current definition of child trie can be only the parentkeyid one, this argument does not really hold).

There is still a case where I find that it is not really good, that would be for child trie that would not be at a location dependant on their type: then the child type would be redundant and would probably use a single value value for this kind of trie and become redundant. But accepting child trie at random location is probably a big redesign with new design concern and a dedicated api could make sense then (and I do not think there is need for such trie).